test/lib/ansible_test/_util/controller/sanity/validate-modules/validate_modules/schema.py

```diff
@@ -47,6 +47,8 @@ def isodate(v, error_code=None):
 
 
 COLLECTION_NAME_RE = re.compile(r'^([^.]+(\.[^.]+)+)$')
+FULLY_QUALIFIED_COLLECTION_RESOURCE_RE = re.compile(r'^([^.]+\.[^.]+(\.[^.]+)+)$')
```

Comment:
Not saying that:
https://github.com/ansible/ansible/blob/devel/lib/ansible/utils/collection_loader/_collection_finder.py#L703
is the authoritative way of doing that, but it certainly looks leaner than above.

```suggestion
COLLECTION_NAME_RE = re.compile(r'^(\w+(\.\w+)+)$')
FULLY_QUALIFIED_COLLECTION_RESOURCE_RE = re.compile(r'^(\w+\.\w{2,})$')
```